# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/DivyankBaluni/ML-/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane and why

I chose **Refresh / Content Opportunity Scoring**.

The goal of this project is to identify which content pages should be reviewed first for possible refreshes or improvements. Instead of checking every page manually, I want to build a ranked list of pages that show evidence of declining performance or missed opportunities based on observable search and engagement signals.

I selected this lane because it directly matches the internship workflow and the starter pipeline while addressing a practical decision that content teams face every day.

In [6]:
# No code required for this section.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*



**Decision**

Which content pages should be reviewed first?

**Who acts?**

A content editor or SEO specialist.

**Action**

The content editor or SEO specialist reviews the highest-ranked pages and decides whether to refresh content, improve CTR, expand information, or investigate other issues.

**Cost of a wrong recommendation**

If an unimportant page is ranked too highly, valuable review time is wasted. If an important declining page is missed, traffic and user engagement may continue to decrease.

**Why ML helps**

Many search and engagement signals interact with each other. A simple rule cannot easily combine all of these signals, while a ranking model can learn useful patterns from historical observations.

In [7]:
# No code required for this section.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [8]:
# !git clone https://github.com/DivyankBaluni/ML_Internship

import pandas as pd

df = pd.read_csv("ML_Internship/data/raw/content_refresh_anonymized.csv")

print("Total pages:", len(df))

declining = (df["trend_direction"] == "down").sum()
print("Declining pages:", declining)

high_visibility = (df["impressions_90d"] >= 500).sum()
print("Pages with >=500 impressions:", high_visibility)

Total pages: 30000
Declining pages: 16262
Pages with >=500 impressions: 16726


The starter dataset contains **30,000** content pages, showing that even the introductory dataset is large enough to require prioritization rather than manual review.

Among these, **16,262** pages are labeled as declining, suggesting that many pages may require attention.

Additionally, **16,726** pages have at least **500 impressions**, indicating that a large number of pages have meaningful visibility. Together, these observations support building a ranking system that helps reviewers decide which pages to investigate first instead of checking every page manually.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

**What I Can and Cannot Claim :**

This project can provide evidence-based, decision-support recommendations using the available search and engagement data.

The rankings produced by the model indicate observed patterns in the historical data and may help prioritize pages for review.

This project cannot prove that updating a page will cause traffic to increase, nor can it predict Google's ranking algorithm. Any findings should be interpreted as observed and directional rather than causal.

In [9]:
# No code required for this section.

## Self-check

Before you submit, confirm each line honestly:

- ✔️ Every section above is filled — markdown thinking AND the code that backs it
- ✔️ The notebook runs top to bottom with no errors (Runtime → Run all)
- ✔️ No client names, URLs, or private queries anywhere
- ✔️ My claims use careful words: observed, measured, directional, decision-support
- ✔️ Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.